File: FunBeads_Pedidos/EnlaceDatos/ConnectionManager.vb
Imports System.Data.SqlClient
Public Class ConnectionManager
    Private sqlConnection As SqlConnection
    Private connectionString As String

    Sub New()
        connectionString = My.Settings.connectionString
        sqlConnection = New SqlConnection
        sqlConnection.ConnectionString = connectionString

    End Sub

    Sub abrir()
        If sqlConnection.State <> ConnectionState.Open Then
            sqlConnection.Open()

        End If

    End Sub

    Sub cerrar()
        If sqlConnection.State <> ConnectionState.Closed Then
            sqlConnection.Close()

        End If

    End Sub

    Function getConnection() As SqlConnection
        Return sqlConnection
    End Function
End Class

File: PracticaExamen/AccesoDatos/ConnectionManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data.SqlClient;

namespace AccesoDatos
{
    public class ConnectionManager
    {
        private String connectionString;
        public SqlConnection connection { get; }

        public ConnectionManager() {
            connectionString = Properties.Settings.Default.connectionString;
            connection = new SqlConnection();
            connection.ConnectionString = this.connectionString;
        }

        public void abrir() {
            if (connection.State != System.Data.ConnectionState.Open)
            {
                connection.Open();
            }
        }

        public void cerrar() {
            if (connection.State != System.Data.ConnectionState.Closed)
            {
                connection.Close();
            }
        }
    }
}
